ocaml/libs/xapi-rrd/lib/rrd.ml

```diff
@@ -22,6 +22,12 @@ exception No_RRA_Available
 
 exception Invalid_data_source of string
 
+(** Inverse is (fun x -> 1.0 - x) *)
+type ds_transform_function = Inverse | Identity
+
+let apply_transform_function f x =
+  match f with Inverse -> 1.0 -. x | Identity -> x
```

Comment:
This works with the current sources, we might want to use max instead of 1.0 (but that has problems when the max is infinite, so we need more considerations here)

It's probably a better idea to change how xen provides stats to provide busy time instead, or the plugin report idle time. I'd prefer the former

Reply:
Or we just transform the data in the new plugin, where we collect the data from xen? Then we don't need complicated generic transform functionality.

Reply:
> Or we just transform the data in the new plugin, where we collect the data from xen? Then we don't need complicated generic transform functionality.

This is worth trying, but it needs precise tracking of time to calculate the difference between the previous read and the current one to be able to calculate the busy time, from the idle time in the current period.

Reply:
I agree that this is an awkward workaround (even though it's implementing the actual specified parameters), just increasing the size of the memory-mapped files (I'm not sure if we could omit `transform: Identity`, taking it as the default, for example, to optimize this). Once I get the timestamps of the plugin to be reliably taken into account, I can drop this and move the transform logic into the plugins.

For now, to accommodate taking inverse from different values, perhaps the variant could take a parameter it would take the inverse from: `Inverse x`?